add new config hashagg_use_magic_hash for tiflash (#20330)
pingcap/docs: tiflash/tiflash-configuration.md

    ## New in v6.4.0. This item is used for the MinTSO scheduler. It specifies the maximum number of queries that can run simultaneously in a TiFlash instance. The default value is 0, which means twice the number of vCPUs. For details about the MinTSO scheduler, see https://docs.pingcap.com/tidb/dev/tiflash-mintso-scheduler.
    task_scheduler_active_set_soft_limit = 0

>     ## New in v9.1.0. TiFlash's HashAgg will use magic hash instead of the default CRC32 if this item is true.

Reviewer: ```suggestion
    ## New in v9.0.0. TiFlash's HashAgg will use magic hash instead of the default CRC32 if this item is true.
```

- This item is used for the MinTSO scheduler. It specifies the maximum number of queries that can run simultaneously in a TiFlash instance. For more information, see [TiFlash MinTSO Scheduler](/tiflash/tiflash-mintso-scheduler.md).
- Default value: `0`, which means twice the number of logical CPU cores

##### `hashagg_use_magic_hash` <span class="version-mark">New in v9.0.0</span>

> - This item is used for the TiFlash Aggregation operator. If this option is set to true, TiFlash's HashAgg will use the magic hash instead of the default CRC32.
> - The results of magic hash have a more uniform distribution, which reduces hash collisions. However, the computation speed will be slower. It is recommended to enable magic hash when the NDV (Number of Distinct Values) of GroupBy key is very high.

Reviewer 2: ```suggestion
- Controls the hash function used by TiFlash during aggregation operations. When set to `true`, TiFlash HashAgg uses the magic hash instead of the default CRC32.
- The magic hash function generates more evenly distributed hash values, reducing hash collisions. However, it is slower than CRC32. It is recommended to enable this configuration when the NDV (number of distinct values) of the `GROUP BY` key is high to optimize performance.
```

- This item is used for the MinTSO scheduler. It specifies the maximum number of queries that can run simultaneously in a TiFlash instance. For more information, see [TiFlash MinTSO Scheduler](/tiflash/tiflash-mintso-scheduler.md).
- Default value: `0`, which means twice the number of logical CPU cores

##### `hashagg_use_magic_hash` <span class="version-mark">New in v9.0.0</span>

> - Controls the hash function used by TiFlash during aggregation operations. When set to `true`, TiFlash HashAgg uses the magic hash instead of the default CRC32.

Reviewer 3: ```suggestion
- Controls the hash function TiFlash uses for aggregation. When set to `true`, TiFlash HashAgg uses the magic hash instead of the default CRC32.
```

> - The magic hash function generates more evenly distributed hash values, reducing hash collisions. However, it is slower than CRC32. It is recommended to enable this configuration when the NDV (number of distinct values) of the `GROUP BY` key is high to optimize performance.

Reviewer 3: ```suggestion
- The magic hash function generates more evenly distributed hash values, reducing hash collisions effectively. However, it is slower than CRC32. It is recommended to enable this configuration when the NDV (number of distinct values) of the `GROUP BY` key is high to optimize aggregation performance.
```
